core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java

```diff
@@ -54,4 +59,36 @@ public UserInfoWrapper getUserInfo(String fiscalCode) {
         return result;
     }
 
+    @Override
+    public UserDetailsWrapper getUserOnboardedProductDetails(String userId, String institutionId, String productId) {
+        log.trace("getUserOnboardedProductDetails start");
+        log.debug("getUserOnboardedProductDetails userId = {}, institutionId = {}, productId = {}", userId, institutionId, productId);
+        UserDetailsWrapper result = null;
+        OnboardingInfoResponse onboardingInfoResponse = msCoreConnector.getInstitutionProductsInfo(userId);
+        Optional<OnboardedInstitutionResponse> institutionResponse = onboardingInfoResponse.getInstitutions().stream()
+                .filter(onboardedInstitutionResponse -> institutionId.equals(onboardedInstitutionResponse.getId())
+                        && productId.equals(onboardedInstitutionResponse.getProductInfo().getId()))
```

Comment:
does api return only product ACTIVE? if there are more than one product you may return a wrong object

Reply:
yes, for example if there are a product DELETED and another product ACTIVE the first will be the DELETED one

Reply:
ah Ok i get it. I just checked and only ACTIVE and Pending Ones are retrieved by default